src/main/kotlin/com/exactpro/th2/kafka/client/Main.kt

```diff
@@ -99,7 +99,7 @@ fun main(args: Array<String>) {
             for (message in batch.messagesList) {
                 LOGGER.trace { "Message ${message.logId} extracted from batch." }
 
-                if (message.metadata.id.bookName != factory.boxConfiguration.bookName) {
+                if (message.metadata.id.bookName?.equals(factory.boxConfiguration.bookName) == true) {
```

Comment:
Looks like we report an error if the book name in metadata is equal to book name in configuration. I think it should work differently